tools/clang/test/CodeGenSPIRV/type.rwbuffer.struct.error.hlsl

```diff
@@ -1,16 +1,16 @@
-// RUN: not %dxc -T vs_6_0 -E main -fcgl  %s -spirv  2>&1 | FileCheck %s
+// RUN: not %dxc -DBUF=RWBuffer                -T vs_6_0 -E main -fcgl  %s -spirv  2>&1 | FileCheck %s -check-prefix=BUF
+// RUN: not %dxc -DBUF=RasterizerOrderedBuffer -T vs_6_0 -E main -fcgl  %s -spirv  2>&1 | FileCheck %s -check-prefix=ROV
 
 struct S {
   float a;
   float b;
 };
 
-// CHECK: error: cannot instantiate RWBuffer with struct type 'S'
-RWBuffer<S> MyRWBuffer;
-
-// CHECK: error: cannot instantiate RasterizerOrderedBuffer with struct type 'S'
-RasterizerOrderedBuffer<S> MyROVBuffer;
+// BUF: error: Elements of typed buffers and textures must be scalars or vectors
+// ROV: error: cannot instantiate RasterizerOrderedBuffer with struct type 'S'
```

Comment:
Shouldn't we produce the same error for ROV as for RWBuffer?

Reply:
Per the checks that you cited in https://github.com/microsoft/DirectXShaderCompiler/blob/d5951b74d76351c17626bf90daf6684ed8598d3d/include/dxc/DXIL/DxilConstants.h#L454, ROV buffers are not considered typed. To convert to using ResKind to determined the typedness, I'd either need to add a check or changed the behavior of IsTyped, which could have implications elsewhere that are not at all related to the intent of this change. I don't mind particularly either way, but I'm not familiar enough with its usage to confidently assert that an ROV buffer is a typed buffer for this or any other purpose.

Reply:
These errors are being produced now.